packages/components/menu/src/menu.ts

```diff
@@ -236,6 +236,8 @@ export default defineComponent({
       }
     )
 
+    watch(items.value, initMenu)
```

Comment:
I think the following code will be more friendly, what do you think
```
watchEffect(() => {
     items.value && initMenu()
 })
```

Reply:
This seems incorrect, since the `initMenu` may also be executed when other deps change. For example, when switching the opening and closing of the sub menu, `initMenu` will be executed so that we can't close the sub menu.

Reply:
Also, I think we no need to invoke the `initMenu ` in these places anymore.

https://github.com/element-plus/element-plus/blob/7da11bc2e6c912a32dad873d90660510284663d4/packages/components/menu/src/menu.ts#L210-L213

https://github.com/element-plus/element-plus/blob/7da11bc2e6c912a32dad873d90660510284663d4/packages/components/menu/src/menu.ts#L292-L297

WDYT?

Reply:
Tested your idea is correct, please wait for more tests

Reply:
Okay, I have updated a commit.

Reply:
well down

Reply:
Thanks for your careful review.